Node/src/TeamsChatConnector.ts

```diff
@@ -155,7 +159,80 @@ export class TeamsChatConnector extends builder.ChatConnector {
           };
           remoteQuery.fetchMemberList(conversationId, options, callback);
         } else {
-          callback(new Error('Failed to authorize request'), null);
+          return callback(new Error('Failed to authorize request'), null);
+        }
+    });
+  }
+
+  /**
+  *  Return a newly started reply chain address in channel
+  *  @param {string} serverUrl - Server url is composed of baseUrl and cloud name, remember to find your correct cloud name in session or the function will not find the team.
+  *  @param {string} channelId - The channel id, will post in the channel.  
+  *  @param {builder.Message|builder.IMessage|builder.IIsMessage} message - The message to post in the channel.
+  *  @param {function} callback - This callback returns err or result.
+  */
+  public startReplyChain(serverUrl: string, channelId: string, message: builder.Message|builder.IMessage|builder.IIsMessage, callback?: (err: Error, address: builder.IChatConnectorAddress) => void) : void {
+    var options: msRest.RequestOptions = {customHeaders: {}, jar: false};
+    var restClient = new RestClient(serverUrl, null);
+    var remoteQuery = new RemoteQuery(restClient);
+    this.getAccessToken((err, token) => {
+        if (!err && token) {
+          options.customHeaders = {
+            'Authorization': 'Bearer ' + token
+          };
+
+          var innerCallback = function (err: Error, result: ReplyResult) {
+            if (result && result.hasOwnProperty("id") && result.hasOwnProperty("activityId"))
+            {
+              var messageAddress = null;
+              if ((<builder.IIsMessage>message).toMessage) {
+                messageAddress = (<builder.IIsMessage>message).toMessage().address;
+              }
+              else if ((<builder.IMessage>message).address)
+              {
+                messageAddress = (<builder.IMessage>message).address;
+              }
+              else
+              {
+                throw new Error("Message type is wrong. Need either Message or IMessage or IIsMessage");
+              }
+
+              var address: builder.IChatConnectorAddress = { ... messageAddress };
+              address.channelId = 'msteams';
+              address.conversation.id = result.id;
+              address.id = result.activityId;
+
+              if (address.user) {
+                  delete address.user;
+              }
+
+              if (callback)
```

Comment:
You could check for callback at the beginning of innerCallback and return early.